src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -189,10 +263,57 @@ private EnvDTE.Project GetProject(string nameOrFileName)
                 => string.Compare(p.FileName, nameOrFileName, StringComparison.OrdinalIgnoreCase) == 0
                 || string.Compare(p.Name, nameOrFileName, StringComparison.OrdinalIgnoreCase) == 0);
 
-        public void AddFile(string projectName, string fileName, string contents = null, bool open = false)
+        public void UpdateOrAddFile(string projectName, string fileName, string contents = null, bool open = false)
```

Comment:
The `UpdateOrAddFile` / `UpdateFile` / `AddFile` methods are all screaming for some doc comments.